tests/system/test_misc.c

```diff
@@ -389,3 +389,88 @@ int cmd_exit_cb(int argc, char **argv)
    DEVSHELL_CMD_ASSERT(after_cb == before_cb + 1);
    return 0;
 }
+
+static bool
+comes_after(struct timeval t2,
+            struct timeval t1)
+{
+   // t1 must come before t2
+   return t2.tv_sec > t1.tv_sec || (t2.tv_sec == t1.tv_sec && t2.tv_usec > t1.tv_usec);
+}
+
+#define BEGIN_OF_TIME (struct timeval) {0, 0}
+
+static void busy_wait(void)
+{
+   // It's assumed that this loop takes
+   // more than 1us of cpu time.
+   for (int i = 0; i < 1000000; i++)
+      asm(""); // Don't let the compiler optimize this loop
```

Comment:
`asm("")` can be optimized by the compiler. Only `asmVolatile` is not optimized. I'd recommend `asmVolatile("nop")` even if, in the future, we'll need to replace that with something more portable.
Also, 1,000,000 of iterations are a lot. We don't need to burn that many cycles for testing something so simple and linear as getrusage(). Can you reduce the number of iterations?